tests/syntax/test_properties.py

```diff
@@ -128,3 +129,157 @@ def test_backRight():
 def test_heading_set_directly():
     with pytest.raises(SpecifierError):
         compileScenic("ego = new Object with heading 4")
+
+
+def test_object_inradius():
+    # Statically Sized Cube Example
+    scenario = compileScenic(
+        """
+        ego = new Object with width 3, with length 3, with height 3,
+            facing (Range(0, 360) deg, Range(0, 360) deg, Range(0, 360) deg)
+        """
+    )
+    ego = sampleEgo(scenario)
+    assert scenario.objects[0].inradius == 1.5
+    assert supportInterval(scenario.objects[0].inradius) == (1.5, 1.5)
+    assert ego.inradius == 1.5
+
+    # Randomly Sized Cube Example
+    scenario = compileScenic(
+        """
+        ego = new Object with width Range(1, 3),
+            with length Range(1, 3), with height Range(1, 3),
+            facing (Range(0, 360) deg, Range(0, 360) deg, Range(0, 360) deg)
+        """
+    )
+    ego = sampleEgo(scenario)
+    assert isinstance(scenario.objects[0].inradius, Distribution)
+    assert supportInterval(scenario.objects[0].inradius) == (0.5, 1.5)
+    assert ego.inradius == pytest.approx(min(ego.width, ego.length, ego.height) / 2)
+
+    # Hollow Static Object Example
+    scenario = compileScenic(
+        """
+        import trimesh
+        hollow_mesh = trimesh.creation.box((1,1,1)).difference(
+            trimesh.creation.box((0.5,0.5,0.5)))
+        ego = new Object with width 3, with length 3, with height 3,
+            facing (Range(0, 360) deg, Range(0, 360) deg, Range(0, 360) deg),
+            with shape MeshShape(hollow_mesh)
+        """
+    )
+    ego = sampleEgo(scenario)
+    assert scenario.objects[0].inradius == 0
+    assert supportInterval(scenario.objects[0].inradius) == (0, 0)
+    assert ego.inradius == 0
+
+    # Hollow Random Object Example
+    scenario = compileScenic(
+        """
+        import trimesh
+        hollow_mesh = trimesh.creation.box((1,1,1)).difference(
+            trimesh.creation.box((0.5,0.5,0.5)))
+        ego = new Object with width Range(1, 3),
+            with length Range(1, 3), with height Range(1, 3),
+            facing (Range(0, 360) deg, Range(0, 360) deg, Range(0, 360) deg),
+            with shape MeshShape(hollow_mesh)
+        """
+    )
+    ego = sampleEgo(scenario)
+    assert isinstance(scenario.objects[0].inradius, Distribution)
```

Comment:
This assertion seems fragile: future improvements might lead to obtaining the constant zero here (from the support interval, we can see that Scenic already "knows" the value has to be zero).

Reply:
Good point, I'll remove this assertion and leave the one above since that really must be a distribution.